### PR TITLE
Implement a ctprobe plugin using the more efficient conntrackprobe

### PR DESCRIPTION
**Pull request by stefanberger**
Implement a ctprobe plugin that is using the more efficient
conntrackprobe fro detecting container traffic.

Signed-off-by: Stefan Berger <stefanb@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by stefanberger** [comment]
For now I am interested in how the tests here end...

File: .travis.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ before_install:
     - docker -v
     # might as well upgrade pip to support TLS and get rid of the warnings
     - sudo -H pip install --upgrade pip
-    - sudo -H pip install codecov pytest-cov
+    - sudo -H pip install codecov pytest-cov==2.3.0 pytest==2.9.2
```
Comment (sahilsuneja1) [inline .travis.yml]: thanks!

File: tests/functional/test_functional_ctprobe.py [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+    class Session(object):
+        def __init__(self, status_code, content):
+            self.status_code = status_code
+            self.content = json.dumps(content)
+
+    return Session(400, {'error': 'Bad request'})
+
+
+# Tests conducted with a single container running.
+class CtprobeFunctionalTests(unittest.TestCase):
```
Comment (sahilsuneja1) [inline tests/functional/test_functional_ctprobe.py]: perfect!

**Comment by sahilsuneja1** [comment]
Although you have great tests, codecov is unhappy coz it bring coverage down from 89 to 88% (see travis output as well). Only exception paths remain untested: https://codecov.io/gh/cloudviz/agentless-system-crawler/pull/303/diff

**Comment by stefanberger** [comment]
Should be close enough now ....

**Comment by sahilsuneja1** [comment]
Indeed, awesome, thanks!

**Pull request merged**